pkgs/desktops/gnome-3/core/gnome-session/default.nix

```diff
@@ -12,6 +12,17 @@ stdenv.mkDerivation rec {
   };
 
   patches = [
+    # Fix debug spam when using systemd session
+    # https://gitlab.gnome.org/GNOME/gnome-session/issues/31
+    (fetchpatch {
+      url = "https://gitlab.gnome.org/GNOME/gnome-session/commit/adfdf7f64f08fc07325f9332e9eba46974cc30ee.patch";
+      sha256 = "0vjg77gpj3k63a5ffhsvv5m30lbj6cab35lhl4gpqxg4j2j3yy7y";
+    })
+    (fetchpatch {
+      url = "https://gitlab.gnome.org/GNOME/gnome-session/commit/b7b24627485c520f873db4e918e217a76ededd8c.patch";
```

Comment:
This patch borrows some code already in gdm that's for unsetting variables.
Is it possible we need to add 
* https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/gnome-3/core/gdm/reset-environment.patch
as well?

Reply:
Not entirely sure. This happens after [`bin/gnome-session`](https://gitlab.gnome.org/GNOME/gnome-session/blob/master/gnome-session/gnome-session.in), which sets up the environment.  So at that point we want `__NIXOS_SET_ENVIRONMENT_DONE` to be propagated (eg. when running shells inside the session).

Reply:
Yeah, I'm unable to reproduce the problem by logging in and out from gnome-shell while keeping `systemd --user` alive.